force-app/main/default/classes/PMT_Payment_TDTM.cls

```diff
@@ -307,39 +307,18 @@ public class PMT_Payment_TDTM extends TDTM_Runnable {
     * @return void Does not return anything
     ********************************************************************************************************/
     private void validatePayments(
-        List<npe01__OppPayment__c> newlist,
-        List<npe01__OppPayment__c> oldlist,
+        List<npe01__OppPayment__c> newList,
+        List<npe01__OppPayment__c> oldList,
         TDTM_Runnable.Action triggerAction,
         DmlWrapper dmlWrapper) {
-        if (triggerAction == TDTM_Runnable.Action.AfterInsert || triggerAction == TDTM_Runnable.Action.AfterUpdate) {
-            for (npe01__OppPayment__c payment : newlist) {
-                if (isEnforceAccountingDataConsistency) {
-                    enforceAccountingDataConsistency(payment);
-                }
-
-                if (payment.npe01__Paid__c == true && payment.npe01__Written_Off__c == true) {
-                    payment.addError(System.Label.pmtWriteOffPaidPayment);
-                }
-            }
-        }
-    }
+            
+        PMT_ValidationService validationService = new PMT_ValidationService(newlist, oldList, triggerAction);
```

Comment:
It looks like this validation executes in all trigger contexts. This should only occur in the `before` context. Does it also need to run in `after`?

Reply:
`validateAccountDataConsistency()` is ran in the After Context.